File: app/src/main/java/dev/ragnarok/fenrir/view/emoji/EmojiconEditText.java
package dev.ragnarok.fenrir.view.emoji;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;

import com.google.android.material.textfield.TextInputEditText;

import dev.ragnarok.fenrir.R;
import dev.ragnarok.fenrir.settings.Settings;

public class EmojiconEditText extends TextInputEditText {

    private int mEmojiconSize;

    public EmojiconEditText(Context context) {
        this(context, null);
    }

    public EmojiconEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs);
    }

    private void init(AttributeSet attrs) {
        mEmojiconSize = (int) getTextSize();

        @SuppressLint("CustomViewStyleable") TypedArray a = getContext().obtainStyledAttributes(attrs, R.styleable.Emojicon);
        mEmojiconSize = (int) a.getDimension(R.styleable.Emojicon_emojiconSize, getTextSize());
        a.recycle();
        setText(getText());
    }

    @Override
    protected void onTextChanged(CharSequence text, int start, int lengthBefore, int lengthAfter) {
        if (!isInEditMode() && !Settings.get().ui().isSystemEmoji()) {
            EmojiconHandler.addEmojis(getContext(), getText(), mEmojiconSize);
        }
    }

    /**
     * Set the size of emojicon in pixels.
     */
    public void setEmojiconSize(int pixels) {
        mEmojiconSize = pixels;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/view/emoji/EmojiconEditText.kt
package dev.ragnarok.fenrir.view.emoji

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import com.google.android.material.textfield.TextInputEditText
import dev.ragnarok.fenrir.EmojiconHandler
import dev.ragnarok.fenrir.R
import dev.ragnarok.fenrir.settings.Settings

class EmojiconEditText @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
    TextInputEditText(
        context, attrs
    ) {
    private var mEmojiconSize = 0
    private fun init(attrs: AttributeSet?) {
        mEmojiconSize = textSize.toInt()
        @SuppressLint("CustomViewStyleable") val a =
            context.obtainStyledAttributes(attrs, R.styleable.Emojicon)
        mEmojiconSize = a.getDimension(R.styleable.Emojicon_emojiconSize, textSize).toInt()
        a.recycle()
        text = text
    }

    override fun onTextChanged(
        text: CharSequence,
        start: Int,
        lengthBefore: Int,
        lengthAfter: Int
    ) {
        if (!isInEditMode && !Settings.get().ui().isSystemEmoji) {
            getText()?.let { EmojiconHandler.addEmojis(context, it, mEmojiconSize) }
        }
    }

    /**
     * Set the size of emojicon in pixels.
     */
    fun setEmojiconSize(pixels: Int) {
        mEmojiconSize = pixels
    }

    init {
        init(attrs)
    }
}